srcCppCommon/PerftestPrinter.cxx

```diff
@@ -0,0 +1,238 @@
+
+
+/*
+ * (c) 2005-2020  Copyright, Real-Time Innovations, Inc. All rights reserved.
+ * Subject to Eclipse Public License v1.0; see LICENSE.md for details.
+ */
+
+
+#include "PerftestPrinter.h"
+
+PerftestOuputFormat get_output_format(std::string outputFormat)
+{
+    if(outputFormat == "legacy"){
+        return LEGACY;
+    } else {
+        return REGULAR;
+    }
+}
+
+PerftestPrinter::PerftestPrinter()
+{
+    _dataLength = 100;
+    _outputFormat = REGULAR;
+    _printSummaryHeaders = true;
+}
+
+void PerftestPrinter::set_header_printed(bool headerPrinted)
+{
+    _headerPrinted = headerPrinted;
+}
+
+void PerftestPrinter::set_print_invertals(bool printIntervals)
+{
+    _printIntervals = printIntervals;
+}
+
+void PerftestPrinter::set_output_format(std::string outputFormat)
+{
+    _outputFormat = get_output_format(outputFormat);
+}
+
+void PerftestPrinter::set_data_length(unsigned int dataLength)
+{
+    if (_outputFormat == LEGACY){
+        printf("\n\n********** New data length is %d\n",
+                dataLength);
+    } else {
+    _dataLength = dataLength;
+    }
+}
+
+void PerftestPrinter::print_pub(unsigned long latency, double latency_ave,
+        double latency_std, unsigned long latency_min,
+        unsigned long latency_max, std::string outputCpu)
+{
+    switch (_outputFormat)
+    {
+        case REGULAR :
+            if (_headerPrinted)
+            {
+                _headerPrinted = !_headerPrinted;
+                printf("\nIntervals One-way Latency for %d Bytes:\n",
```

Comment:
I would not print this here, but before, the reason is that if you are sending very big samples, you might need to wait a while until you receive this sample.

Reply:
See my comment above